src/compile/legend.ts

```diff
@@ -106,6 +110,17 @@ export function values(legend: Legend) {
   return vals;
 }
 
+export function type(legend: Legend, fieldDef: FieldDef, channel: Channel) {
+  if (legend.type) {
+    return legend.type;
+  }
+
+  if (channel === COLOR && !fieldDef.bin && !fieldDef.timeUnit && contains([QUANTITATIVE, TEMPORAL], fieldDef.type)) {
```

Comment:
Hmm, you are right. I was thinking that it's similar to binning but I guess we sometimes have quite a large number of bins.